Add subproject_options documentation (#3655)
conan-io/docs: reference/tools/meson/mesontoolchain.rst

Note that in this case, Meson might be able to find dependencies in "wraps", it is the responsibility of the user to check the behavior and make sure about the dependencies origin.

subproject_options
^^^^^^^^^^^^^^^^^^

This attribute allows defining Meson subproject options:

.. code:: python

    def generate(self):
        tc = MesonToolchain(self)
        tc.subproject_options["SUBPROJECT"] = [{'MYVAR': 'MyValue'}]
        tc.generate()

This is translated to:

> - One subproject ``SUBPROJECT`` along with options definition for ``MYVAR`` in ``conan_meson_native.ini`` or ``conan_meson_cross.ini`` file.

Reviewer: ```suggestion
- One subproject ``SUBPROJECT`` and option definition for ``MYVAR`` in the ``conan_meson_native.ini`` or ``conan_meson_cross.ini`` file.
```

> Note that in contrast to project_options, subproject_options is a dictionary of lists of dictionaries. This is because Meson allows multiple subprojects, and each subproject can have multiple options.

Reviewer: ```suggestion
Note that in contrast to ``project_options``, ``subproject_options`` is a dictionary of lists of dictionaries. This is because Meson allows multiple subprojects, and each subproject can have multiple options.
```
